includes/common.php

```diff
@@ -1527,9 +1527,12 @@ function load_os(&$device)
     );
 
     // Set type to a predefined type for the OS if it's not already set
-    if ($device['type'] == 'unknown' || $device['type'] == '') {
+    if ($device['type'] == 'unknown' || $device['type'] == '' || $config['os'][$device['os']]['type'] != $device['type']) {
```

Comment:
It feels like this should be, I'm not somewhere where I can test it right now...
```php
    // Set type to a predefined type for the OS if it's not already set
    if ($config['os'][$device['os']]['type'] && $device['type'] != $config['os'][$device['os']]['type']) {
        $device['type'] = $config['os'][$device['os']]['type'];
        $sql = dbUpdate(array('type' => $device['type']), 'devices', 'device_id=?', array($device['device_id']));
        log_event('Device type changed '.$device['type'].' => '.$config['os'][$device['os']]['type'], $device, 'system');
        echo "Changed Type! : ".$device['type'].PHP_EOL;
    }
```